persist-cli/src/main/java/io/ballerina/persist/cmd/Pull.java

```diff
@@ -0,0 +1,242 @@
+public class Pull implements BLauncherCmd {
+    private static final PrintStream errStream = System.err;
+
+    private final String sourcePath;
+
+    private static final String COMMAND_IDENTIFIER = "persist-pull";
+
+    DatabaseConnector databaseConnector;
+
+    public Pull() {
+        this("");
+    }
+
+    public Pull(String sourcePath) {
+        this.sourcePath = sourcePath;
+    }
+
+    @CommandLine.Option(names = {"--datastore"})
+    private String datastore = "mysql";
+
+    @CommandLine.Option(names = {"--host"})
+    private String host;
+
+    @CommandLine.Option(names = {"--port"})
+    private String port = "3306";
+
+    @CommandLine.Option(names = {"--user"})
+    private String user;
+
+    @CommandLine.Option(names = {"--database"})
+    private String database;
+
+    @CommandLine.Option(names = { "-h", "--help" }, hidden = true)
+    private boolean helpFlag;
+
+    @Override
+    public void execute() {
+        Scanner scanner = new Scanner(System.in, StandardCharsets.UTF_8);
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
+            errStream.println(commandUsageInfo);
+            return;
+        }
+        try {
+            validatePullCommandOptions(datastore, host, port, user, database);
+        } catch (BalException e) {
+            errStream.println("Invalid Option(s): \n" + e.getMessage());
+            return;
+        }
+
+        String password = readDatabasePassword(scanner, errStream);
+
+        try {
+            validateBallerinaProject(Paths.get(this.sourcePath));
+        } catch (BalException e) {
+            errStream.println(e.getMessage());
+            return;
+        }
+
+        if (this.datastore.equals(PersistToolsConstants.SupportedDataSources.MYSQL_DB)) {
+            this.databaseConnector = new DatabaseConnector(JDBC_URL_WITH_DATABASE_MYSQL, MYSQL_DRIVER_CLASS);
+        } else {
+            errStream.printf("ERROR: unsupported data store: '%s'%n", datastore);
+            return;
+        }
+
+        Path persistDir = Paths.get(this.sourcePath, PERSIST_DIRECTORY);
+        if (!Files.exists(persistDir)) {
+            try {
+                Files.createDirectory(persistDir.toAbsolutePath());
+            } catch (IOException e) {
+                errStream.println("ERROR: failed to create the persist directory. " + e.getMessage());
+                return;
+            }
+        }
+
+        boolean modelFile = Files.exists(Path.of(String.valueOf(persistDir), "model.bal"));
+        if (modelFile) {
+            String yellowColor = "\u001B[33m";
+            String resetColor = "\u001B[0m";
+            errStream.print(yellowColor + "A model.bal file already exists. " +
+                    "Continuing would overwrite it. Do you wish to continue? (y/n) " + resetColor);
+            String input = scanner.nextLine();
+            if (!input.toLowerCase(Locale.ENGLISH).equals("y")) {
```

Comment:
Just wondering if should we also consider the input `yes` here. WDYT?

Reply:
+1. I will change this in the next commit.

Reply:
This was fixed in https://github.com/ballerina-platform/persist-tools/pull/330/commits/aed153000919d8a746bfa6a8fc3f76281ea83789